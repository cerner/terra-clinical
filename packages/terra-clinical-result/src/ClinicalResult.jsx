import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import classNames from 'classnames/bind';
import ClinicalResultDisplay from './_ClinicalResultDisplay';
import ResultError from './common/other/_ResultError';
import NoData from './common/other/_KnownNoData';
import statusPropType from './proptypes/statusPropTypes';
import interpretationPropType from './proptypes/interpretationPropTypes';
import valueQuantityPropType from './proptypes/valuePropTypes';
import styles from './ClinicalResult.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * A unique event identifier attached to the result data
   */
  eventId: PropTypes.string,
  /**
   *  Value and optional Unit of Measure for the Observation Result
   */
  result: valueQuantityPropType,
  /**
   * Interpretation of the Result, indicates Criticality
   */
  interpretation: interpretationPropType,
  /**
   * If Result has not been verified
   */
  isUnverified: PropTypes.bool,
  /**
   *  Display to show the full Result Name/Label Concept, e.g. `'Temperature Oral'`.
   */
  conceptDisplay: PropTypes.string,
  /**
   *  Display to show an appropriate clinically relevant documented datetime.
   */
  datetimeDisplay: PropTypes.string,
  /**
   * Visually hides the unit of measure when presented in a series of side-by-side columns of the same unit.
   */
  hideUnit: PropTypes.bool,
  /**
   * Whether or not the text should be truncated in display. Restricts clinical result details each to one line.
   */
  isTruncated: PropTypes.bool,
  /**
   *  If the Result value has not been authenticated and committed to patient chart.
   */
  isModified: PropTypes.bool,
  /**
   *  If the Result value has been modified from it's original value for the same clinically documented event & datetime.
   */
  hasComment: PropTypes.bool,
  /**
   * Override that shows an Error display. Used when there is a known error or problem when retrieving or assembling the clinical result data.
   */
  hasResultError: PropTypes.bool,
  /**
   * Override that shows a known "No Data" display. Used when there is known to be no value for a given clinical result concept at a specific datetime.
   */
  hasResultNoData: PropTypes.bool,
  /**
   * Enum for possible Result Statuses.
   * One of `'entered-in-error'`s.
   */
  status: statusPropType,
  /**
   * @private
   * Used by Flowsheet Result Cell to hide icons because it displays them in different positions.
   */
  hideAccessoryDisplays: PropTypes.bool,
  /**
   * @private
   * The intl object to be injected for translations.
   */
  intl: intlShape.isRequired,
};

const defaultProps = {
  result: {},
};

const ClinicalResult = (props) => {
  const {
    eventId,
    result,
    interpretation,
    hideUnit,
    isTruncated,
    isUnverified,
    isModified,
    hasComment,
    hasResultError,
    hasResultNoData,
    hideAccessoryDisplays,
    conceptDisplay,
    status,
    datetimeDisplay,
    ...customProps
  } = props;

  let clinicalResultDisplay;

  if (hasResultError) {
    clinicalResultDisplay = <ResultError />;
  } else if (hasResultNoData) {
    clinicalResultDisplay = <NoData />;
  } else {
    clinicalResultDisplay = (
      <ClinicalResultDisplay
        eventId={eventId}
        result={result}
        interpretation={interpretation}
        isUnverified={isUnverified}
        status={status}
        hideUnit={hideUnit}
        isTruncated={isTruncated}
        hideAccessoryDisplays={hideAccessoryDisplays}
        isModified={isModified}
        hasComment={hasComment}
        conceptDisplay={conceptDisplay}
        datetimeDisplay={datetimeDisplay}
      />
    );
  }

  const clinicalResultClassnames = cx([
    'clinical-result',
    { truncated: isTruncated },
  ]);

  return (
    <div
      {...customProps}
      className={customProps.className ? `${clinicalResultClassnames} ${customProps.className}` : clinicalResultClassnames}
    >
      {clinicalResultDisplay}
    </div>
  );
};

ClinicalResult.propTypes = propTypes;
ClinicalResult.defaultProps = defaultProps;

export default injectIntl(ClinicalResult);
