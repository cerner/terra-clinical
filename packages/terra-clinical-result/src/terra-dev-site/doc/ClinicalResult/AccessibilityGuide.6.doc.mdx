import { Badge } from 'terra-clinical-result/package.json?dev-site-package';
import { Notice } from "@cerner/terra-docs";
import ClinicalResultBloodPressureExtraDisplays from '../example/clinical-result-blood-pressure/ClinicalResultBloodPressureExtraDisplays';
import ClinicalResultKnownError from '../example/clinical-result/ClinicalResultKnownError';
import FlowsheetResultCellMultipleResultsSymbols from '../example/flowsheet-result-cell/FlowsheetResultCellMultipleResultsSymbols?dev-site-example';
import ResultNameHeaderCellAccessibleNamesExample from '../example/result-name-header-cell/ResultNameHeaderCellAccessibleNamesExample';

<Badge />

# Accessibility Guide for Terra Clinical Result

## Why is this important?

Terra Clinical Result is a collection of standardized views for single result values added to a patient's medical chart. As there are many patients who rely on assistive technologies to interact with their chart, it is critical that any information conveyed by a clinical result to sighted users is also conveyed to screen reader users in a meaningful way.

## Accessibility Considerations

Terra Clinical Result has several variants designed to either display specific results or for use within a flowsheet. The table below lists each variant and its use:

| Variant | Use |
| --- | --- |
| Clinical Result | The default Clinical Result component intended for displaying a single result value |
| Clinical Result Blood Pressure | Intended for displaying Blood Pressure Vital Sign results |
| Flowsheet Result Cell | Specifically for the grid display of a result collection |
| Result Name Header Cell | Standardized view for the Result Name and Unit of Measure in the row header for each row within a flowsheet |
| Result Time Header Cell | Standardized view for the Date/Time Header cells for each bucketed time column within a flowsheet |

The following sections go into more detail on accessibility considerations for each variant.

----

### Code Considerations

#### Clinical Result Blood Pressure with Additional Displays

Clinical Result Blood Pressure creates a standardized view for single result values added to a patient's medical chart. Along with the blood pressure results, we have the option to display additional details like concept names (Systolic, Diastolic) and the date and time of the results. Visually we can see that this information is related but it is possible that screen reader users may not understand the whole cell as a group.

<Notice variant="important" ariaLevel="5">

It is highly recommended to set the optional `isBloodPressureGrouped` [prop](https://engineering.cerner.com/terra-ui/components/terra-clinical-result/clinical-result/clinical-result-blood-pressure#clinical-result-blood-pressure-props) to true when additional information is displayed with the blood pressure results. This will semantically relate the blood pressure results and additional displays as a group context.

**Code Example:**

<div aria-label="Example demo">
  <ClinicalResultBloodPressureExtraDisplays />
</div>
<div aria-label="Example code">

    import React from 'react';
    import { ClinicalResultBloodPressure } from 'terra-clinical-result/lib';

    const bloodpressureResultValue = {
      id: '111',
      systolic: {
        eventId: '111.1',
        result: {
          value: '120',
          unit: 'mmHg',
        },
        conceptDisplay: 'Systolic',
        datetimeDisplay: 'Nov 23, 2019 13:31:31',
      },
      diastolic: {
        eventId: '111.2',
        result: {
          value: '80',
          unit: 'mmHg',
        },
        conceptDisplay: 'Diastolic',
        datetimeDisplay: 'Nov 23, 2019 13:31:31',
      },
      isBloodPressureGrouped: true,
    };

    export default () => <ClinicalResultBloodPressure {...bloodpressureResultValue} />;

</div>

This example shows how to set up the optional `isBloodPressureGrouped` prop in order to group the Blood Pressure results with additional displays.

</Notice>

#### Result Name Header Cell Accessible Names

Result Name Header Cell creates a standardized view for the Result Name and Unit of Measure pair within a flowsheet. In many cases, the displayed result name or displayed unit of measure is an abbreviation. Reading out these abbreviations is not very helpful for screen reader users.

<Notice variant="important" ariaLevel="5">

In order to provide better accessible names for the result name and unit of measure, the optional `fullResultName` and `fullUnit` [props](https://engineering.cerner.com/terra-ui/components/terra-clinical-result/clinical-result/result-name-header-cell#result-name-header-cell-props) can be used to set the full name of the result name and unit when consuming the Result Name Header Cell component.

**Code Example:**

<div aria-label="Example demo">
  <ResultNameHeaderCellAccessibleNamesExample />
</div>
<div aria-label="Example code">

    import React from 'react';
    import { ResultNameHeaderCell } from 'terra-clinical-result/lib/index';
    import classNames from 'classnames';
    import classNamesBind from 'classnames/bind';
    import ThemeContext from 'terra-theme-context';
    import styles from '../Examples.module.scss';

    const cx = classNamesBind.bind(styles);

    const ResultNameHeaderCellAccessibleNamesExample = () => {
      const theme = React.useContext(ThemeContext);
      const mockFlowsheetClassnames = classNames(
        cx(
          'mock-flowsheet-example',
          theme.className,
        ),
      );

      return (
        <div className={mockFlowsheetClassnames}>
          <div className={cx('mock-flowsheet-firstcolumn')}>
            <div className={cx('mock-flowsheet-firstcolumn-cell')}>
              <ResultNameHeaderCell resultName="Temp" fullResultName="Temperature" unit="degC" fullUnit="Degrees Celsius" />
            </div>
          </div>
        </div>
      );
    };

    export default ResultNameHeaderCellAccessibleNamesExample;

</div>

This example shows how to set up the optional `fullResultName` and `fullUnit` props in order to provide accessible names for the displayed result name and unit.

</Notice>

#### Accessibility of Cell Components

Flowsheet Result Cell, Result Name Header Cell, and Result Time Header Cell all return their content using HTML table elements. Flowsheet Result Cell utilizes the `<td>` element while both of the header cell components utilize the `<th>` element. As such, the best way to consume these components for true accessibility is in a semantic HTML table. Otherwise, the context of being a cell/header cell will not be understood by screen reader users. The [Terra HTML Table](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/html-table/about) component can be used to create a semantic HTML table for this purpose.

### Other Considerations

#### Flowsheet Result Cell with Multiple Results

Flowsheet Result Cell is used for the grid display of a result collection and can handle displaying an indication of multiple results. When there are multiple results, the first result is displayed along with a custom symbol that is used to communicate information about the remaining results:

<FlowsheetResultCellMultipleResultsSymbols />

The number represents how many additional results exist and the color of the result stack indicates the severity of the **most** serious result in the side panel.

Alternative text has been added to communicate the number of additional results and the severity:

| Severity Color | Meaning | Text to be Read |
| --- | --- | --- |
| Gray | Normal | "A total of [number] results are available." |
| Light Red | Out of range (high or low) but not critical | "A total of [number] results are available with at least 1 out of range result." |
| Dark Red | Critical | "A total of [number] results are available with at least 1 critical result." |

e.g. if we have a dark red box with 3 results the screenreader will read out:

“A total of 3 results are available with at least 1 critical result.”

#### Usage of Visually Hidden Text

There are a few cases in which we leverage visually hidden text to give greater context to certain styling on the front end - or related to statuses of certain results. These cases are as follows:

##### No Results:

When the `hasResultNoData` prop is used the display will show: '--'

Hidden text has been added to read out "No Results" when the `hasResultNoData` prop is used.

##### Entered In Error:

When a result inside a Flowsheet Result Cell has a status of `entered-in-error`, the display will show: 'In Error'

Hidden text has been added to read out "Value Uncharted as In Error" to convey this has a status of `entered-in-error`.

##### Strikethrough Result Entered In Error:

When a result is shown on the front end with a strikethrough, that means it has been entered in error. The result is important enough to be shown visually and as such, needs to be read by the screen reader while also giving some context that it has a status of `entered-in-error`.

Hidden text has been added to read out "Result {strikethroughResult} Entered In Error" when we have a strikethrough result.

## Support Compliance

Terra is committed to ensuring its components provide maximum possible accessibility. However, simply using Terra components will not automatically make a product accessible.

Final responsibility lies with the consumers of Terra components &mdash; ensuring proper usage, engineers following correct implementation patterns, and authors crafting content that follows best practice guidance &mdash; to make a product fully accessible.

### WCAG Resources

#### Success Criteria

- [**1.1.1 Non-text Content**](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content) - All non-text content that is presented to the user has a text alternative that serves the equivalent purpose.
- [**1.3.1 Info and Relationships**](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships) - Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text. (Level A)
- [**3.2.4 Consistent Identification**](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification) - Components that have the same functionality within a set of Web pages are identified consistently.
- [**4.1.2 Name, Role, Value**](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value) - For all user interface components (including but not limited to: form elements, links and components generated by scripts), the name and role can be programmatically determined; states, properties, and values that can be set by the user can be programmatically set; and notification of changes to these items is available to user agents, including assistive technologies.

### Partial Support & Requiring Further Enhancement

- None identified
- [Report a problem with this component on **GitHub**](https://github.com/cerner/terra-clinical/issues/new/choose)

_For more information about Accessibility Support with Terra — go to [Component Standards: Accessibility (A11y)](https://engineering.cerner.com/terra-ui/about/terra-ui/component-standards#accessibility-a11y)._
