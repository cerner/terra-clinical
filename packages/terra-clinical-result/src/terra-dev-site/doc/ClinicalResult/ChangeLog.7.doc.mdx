import { Badge } from 'terra-clinical-result/package.json?dev-site-package';

import ChangeLog from 'terra-clinical-result/CHANGELOG.md';

<Badge />

<ChangeLog />
