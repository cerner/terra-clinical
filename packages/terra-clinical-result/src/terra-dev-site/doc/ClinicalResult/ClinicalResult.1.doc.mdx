import { Badge } from 'terra-clinical-result/package.json?dev-site-package';

import DefaultResult from '../example/clinical-result/ClinicalResultDefault?dev-site-example';
import DecoratedResult from '../example/clinical-result/ClinicalResultDecorated?dev-site-example';
import UnverifiedResult from '../example/clinical-result/ClinicalResultUnverified?dev-site-example';
import ExtraDisplaysResult from '../example/clinical-result/ClinicalResultExtraDisplays?dev-site-example';
import TruncatedResult from '../example/clinical-result/ClinicalResultTruncated?dev-site-example';
import InterpretationResults from '../example/clinical-result/ClinicalResultInterpretation?dev-site-example';
import SizeChangeResult from '../example/clinical-result/ClinicalResultChangeSize?dev-site-example';
import KnownNoData from '../example/clinical-result/ClinicalResultKnownNoData?dev-site-example';
import KnownError from '../example/clinical-result/ClinicalResultKnownError?dev-site-example';
import AccidentalError from '../example/clinical-result/ClinicalResultAccidentalError?dev-site-example';
import StatusEnteredInError from '../example/clinical-result/ClinicalResultStatusEnteredInError?dev-site-example';

import ClinicalResultsProps from 'terra-clinical-result/src/ClinicalResult?dev-site-props-table';

<Badge />
# Terra Clinical Result

The Terra Clinical Result package is a collection of standardized views for presenting clinical results documented to a Patient's Medical Chart,
such as Vital Signs, Laboratory Results, and Discretely Documented Values. It also includes additional supporting views where clinical results
are commonly displayed like flowsheets, result detail viewers, etc.

This default Clinical Result component is intended for displaying a single result value.

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-clinical-result`

<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
## Peer Dependencies

This component requires the following peer dependencies be installed in your app for the component to properly function.

| Peer Dependency | Version |
|-|-|
| react | ^16.8.5 |
| react-dom | ^16.8.5 |
| react-intl | ^2.8.0 |

<!-- AUTO-GENERATED-CONTENT:END -->

## Usage Guide
The clinical result is constructed by providing structured object with a specific construction that follows a similar pattern to the [HL7 FHIR Observation](https://www.hl7.org/fhir/observation.html) standard.

An example of a single clinical result value:
```jsx
import ClinicalResult from 'terra-clinical-result/lib/index';
```

## Component Features

* [Cross-Browser Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#cross-browser-support)
* [Responsive Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#responsive-support)
* [Mobile Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#mobile-support)
* [LTR/RTL Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#ltr--rtl-support)


[View component source code](https://github.com/cerner/terra-clinical/tree/master/packages/terra-clinical-result)

## Examples
<DefaultResult title="Default Clinical Result" />
<DecoratedResult title="Modifiied Clinical Result with Comments" />
<UnverifiedResult title="Unverified Clinical Result with Hide Unit of Measure" />
<ExtraDisplaysResult title="Clinical Result with additional displays" />
<InterpretationResults title="Clinical Results with Intrepretation" />
<SizeChangeResult title="Changing the Clinical Result Size" />
<TruncatedResult title="Truncated Clinical Result" />
<KnownNoData title="Display for 'Known No Result'" />
<KnownError title="Clinical Result Known Error" />
<AccidentalError title="Clinical Result Accidental Error" />
<StatusEnteredInError title="Clinical Result Status Entered In Error" />

## Clinical Result Props Props
<ClinicalResultsProps />