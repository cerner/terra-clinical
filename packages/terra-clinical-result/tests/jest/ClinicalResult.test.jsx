import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import ClinicalResult from '../../src/ClinicalResult';
import DefaultResult from '../../src/terra-dev-site/test/clinical-result/TestResults';

describe('ClinicalResult', () => {
  it('should render a default ClinicalResult', () => {
    const result = shallowWithIntl(<ClinicalResult {...DefaultResult} />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResult', () => {
    const result = shallowWithIntl(<ClinicalResult {...DefaultResult} isTruncated />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResult with icons moved', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallowWithIntl(<ClinicalResult {...resultData} isTruncated isModified />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should pass certain props to Observation ', () => {
    const resultData = {
      ...DefaultResult,
      interpretation: 'critical',
    };
    const result = shallowWithIntl(<ClinicalResult {...resultData} hideUnit />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a modified result ', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallowWithIntl(<ClinicalResult {...resultData} isModified />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a commented result ', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallowWithIntl(<ClinicalResult {...resultData} hasComment />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render an unverified result ', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallowWithIntl(<ClinicalResult {...resultData} isUnverified />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a result with unit hidden', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallowWithIntl(<ClinicalResult {...resultData} hideUnit />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a result with a concept display', () => {
    const resultData = {
      ...DefaultResult,
      conceptDisplay: 'Temperature Oral',
    };
    const result = shallowWithIntl(<ClinicalResult {...resultData} />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a result with a date time display', () => {
    const resultData = {
      ...DefaultResult,
      datetimeDisplay: 'Nov 23, 2019 13:31:31',
    };
    const result = shallowWithIntl(<ClinicalResult {...resultData} />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a entered-in-error status result', () => {
    const resultData = {
      ...DefaultResult,
      status: 'entered-in-error',
      interpretation: 'critical',
    };
    const result = shallowWithIntl(<ClinicalResult {...resultData} />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a ResultError if hasResultError is true', () => {
    const result = shallowWithIntl(<ClinicalResult hasResultError />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a NoData if hasResultNoData is true', () => {
    const result = shallowWithIntl(<ClinicalResult hasResultNoData />).dive();
    expect(result).toMatchSnapshot();
  });
});
