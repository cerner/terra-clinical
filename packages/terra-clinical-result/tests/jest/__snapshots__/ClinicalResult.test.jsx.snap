// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClinicalResult should pass certain props to Observation  1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    eventId="111"
    hideUnit={true}
    interpretation="critical"
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;

exports[`ClinicalResult should render a NoData if hasResultNoData is true 1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <InjectIntl(KnownNoData) />
</div>
`;

exports[`ClinicalResult should render a ResultError if hasResultError is true 1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <InjectIntl(ResultError) />
</div>
`;

exports[`ClinicalResult should render a commented result  1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    eventId="111"
    hasComment={true}
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;

exports[`ClinicalResult should render a default ClinicalResult 1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    eventId="111"
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;

exports[`ClinicalResult should render a entered-in-error status result 1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    eventId="111"
    interpretation="critical"
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
    status="entered-in-error"
  />
</div>
`;

exports[`ClinicalResult should render a modified result  1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    eventId="111"
    isModified={true}
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;

exports[`ClinicalResult should render a result with a concept display 1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    conceptDisplay="Temperature Oral"
    eventId="111"
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;

exports[`ClinicalResult should render a result with a date time display 1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    datetimeDisplay="Nov 23, 2019 13:31:31"
    eventId="111"
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;

exports[`ClinicalResult should render a result with unit hidden 1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    eventId="111"
    hideUnit={true}
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;

exports[`ClinicalResult should render a truncated ClinicalResult 1`] = `
<div
  className="clinical-result truncated"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    eventId="111"
    isTruncated={true}
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;

exports[`ClinicalResult should render a truncated ClinicalResult with icons moved 1`] = `
<div
  className="clinical-result truncated"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    eventId="111"
    isModified={true}
    isTruncated={true}
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;

exports[`ClinicalResult should render an unverified result  1`] = `
<div
  className="clinical-result"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <ClinicalResultDisplay
    eventId="111"
    isUnverified={true}
    result={
      Object {
        "unit": "mL",
        "value": "12345.678",
      }
    }
  />
</div>
`;
